tests/unit_tests/gui/simulation/view/test_legend.py

```diff
@@ -1,16 +1,75 @@
-from PyQt5.QtCore import QModelIndex
+from random import randint
 
-from ert.gui.model.progress_proxy import ProgressProxyModel
-from ert.gui.model.snapshot import SnapshotModel
-from ert.gui.simulation.view.legend import LegendView
+from qtpy.QtWidgets import QLabel
 
+from ert.ensemble_evaluator.state import REAL_STATE_TO_COLOR
+from ert.gui.simulation.view import ProgressWidget
+from tests.unit_tests.gui.conftest import get_child
 
-def test_delegate_instantiated(qtbot, large_snapshot):
-    snapshot_model = SnapshotModel()
-    progress_proxy_model = ProgressProxyModel(snapshot_model)
-    snapshot_model._add_snapshot(SnapshotModel.prerender(large_snapshot), 0)
 
-    legend_view = LegendView()
-    qtbot.addWidget(legend_view)
-    legend_view.setModel(progress_proxy_model)
-    assert legend_view.itemDelegate(QModelIndex())
+def test_progress_step_changing(qtbot):
+    progress_widget = ProgressWidget()
+    qtbot.addWidget(progress_widget)
+
+    realization_count = 1
+
+    for i in range(len(REAL_STATE_TO_COLOR.keys())):
+        status = {}
+
+        # generate new list with one flag set
+        for u, state in enumerate(REAL_STATE_TO_COLOR.keys()):
+            status[state] = 1 if i == u else 0
+
+        progress_widget.update_progress(status, realization_count)
+
+        for state in REAL_STATE_TO_COLOR:
+            label_marker = get_child(
+                progress_widget,
+                QLabel,
+                name=f"progress_label_text_{state}",
+            )
+
+            assert label_marker
+            count = status[state]
+            assert f" {state} ({count}/{realization_count})" in label_marker.text()
+
+
+def test_progress_state_width_correct(qtbot):
+    progress_widget = ProgressWidget()
+    qtbot.addWidget(progress_widget)
+    status = {"Unknown": 20}
+    realization_count = 20
+    progress_widget.update_progress(status, realization_count)
+
+    progress_marker = get_child(
+        progress_widget,
+        QLabel,
+        name="progress_Unknown",
+    )
+
+    assert progress_marker
+    base_width = progress_marker.width() / realization_count
+
+    spread = realization_count
+    gen_list = []
+
+    for _ in range(len(REAL_STATE_TO_COLOR)):
+        r = randint(0, spread)
+        gen_list.append(r)
+        spread -= r
+
+    for i, state in enumerate(REAL_STATE_TO_COLOR.keys()):
+        status[state] = gen_list[i]
```

Comment:
Could use hypothesis here to generate input.
We just need to retain some sensical sum for the amount of realizations. Once this shows unrealistically high numbers, the visuals will not add up perfectly. For instance if the number of reals is way more than number of pixels available.

Reply:
Came up with this 
```python
from hypothesis import given, settings, HealthCheck
import pytest
import hypothesis.strategies as st

from qtpy.QtWidgets import QLabel

from ert.ensemble_evaluator.state import REAL_STATE_TO_COLOR
from ert.gui.simulation.view import ProgressWidget
from tests.unit_tests.gui.conftest import get_child


def test_progress_step_changing(qtbot):
    progress_widget = ProgressWidget()
    qtbot.addWidget(progress_widget)

    realization_count = 1

    for i in range(len(REAL_STATE_TO_COLOR.keys())):
        status = {}

        # generate new list with one flag set
        for u, state in enumerate(REAL_STATE_TO_COLOR.keys()):
            status[state] = 1 if i == u else 0

        progress_widget.update_progress(status, realization_count)

        for state in REAL_STATE_TO_COLOR:
            label_marker = get_child(
                progress_widget,
                QLabel,
                name=f"progress_label_text_{state}",
            )

            assert label_marker
            count = status[state]
            assert f" {state} ({count}/{realization_count})" in label_marker.text()


@given(
    realization_count=st.integers(min_value=3, max_value=100),
    status=st.lists(st.sampled_from(list(REAL_STATE_TO_COLOR.keys()))),
)
@settings(suppress_health_check=[HealthCheck.function_scoped_fixture])
def test_progress_state_width_correct(qtbot, status, realization_count):
    progress_widget = ProgressWidget()
    qtbot.addWidget(progress_widget)
    status = {"Unknown": realization_count}
    progress_widget.update_progress(status, realization_count)

    progress_marker = get_child(
        progress_widget,
        QLabel,
        name="progress_Unknown",
    )

    assert progress_marker
    base_width = progress_marker.width() / realization_count
    progress_widget.update_progress(status, realization_count)

    for state in REAL_STATE_TO_COLOR:
        progress_marker = get_child(
            progress_widget,
            QLabel,
            name=f"progress_{state}",
        )

        assert progress_marker
        count = status.get(state, 0)
        assert progress_marker.width() == pytest.approx(base_width * count)

```

Reply:
After talking we found this: 

```python
from hypothesis import given, settings, HealthCheck
from typing import Dict
import pytest
import hypothesis.strategies as st

from qtpy.QtWidgets import QLabel

from ert.ensemble_evaluator.state import REAL_STATE_TO_COLOR
from ert.gui.simulation.view import ProgressWidget
from tests.unit_tests.gui.conftest import get_child


@given(realization_count=st.integers(min_value=3, max_value=100))
@settings(suppress_health_check=[HealthCheck.function_scoped_fixture])
def test_progress_step_changing(qtbot, realization_count):
    progress_widget = ProgressWidget()
    qtbot.addWidget(progress_widget)

    for i in range(len(REAL_STATE_TO_COLOR.keys())):
        status = {}

        # generate new list with one flag set
        for u, state in enumerate(REAL_STATE_TO_COLOR.keys()):
            status[state] = 1 if i == u else 0

        progress_widget.update_progress(status, realization_count)

        for state in REAL_STATE_TO_COLOR:
            label_marker = get_child(
                progress_widget,
                QLabel,
                name=f"progress_label_text_{state}",
            )

            assert label_marker
            count = status[state]
            assert f" {state} ({count}/{realization_count})" in label_marker.text()


@given(
    status=st.dictionaries(
        st.sampled_from(list(REAL_STATE_TO_COLOR.keys())),
        st.integers(min_value=1, max_value=10),
        min_size=1,
    )
)
@settings(suppress_health_check=[HealthCheck.function_scoped_fixture])
def test_progress_state_width_correct(qtbot, status: Dict[str, int]):
    realization_count = sum(status.values())
    progress_widget = ProgressWidget()
    qtbot.addWidget(progress_widget)
    status = {"Unknown": realization_count}
    progress_widget.update_progress(status, realization_count)

    progress_marker = get_child(
        progress_widget,
        QLabel,
        name="progress_Unknown",
    )

    assert progress_marker
    base_width = progress_marker.width() / realization_count
    progress_widget.update_progress(status, realization_count)

    for state in REAL_STATE_TO_COLOR:
        progress_marker = get_child(
            progress_widget,
            QLabel,
            name=f"progress_{state}",
        )

        assert progress_marker
        count = status.get(state, 0)
        assert progress_marker.width() == pytest.approx(base_width * count)
```